api/envoy/admin/v2alpha/clusters.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package envoy.admin.v2alpha;
+
+import "envoy/api/v2/core/base.proto";
+import "envoy/api/v2/core/address.proto";
+import "envoy/type/percent.proto";
+
+// [#protodoc-title: Clusters]
+
+// Admin endpoint uses this wrapper for `/clusters` to display cluster status information.
+// See :ref:`/clusters <operations_admin_interface_clusters>` for more information.
+message Clusters {
+  // Mapping from cluster name to each cluster's status.
+  repeated ClusterStatus cluster_statuses = 1;
+}
+
+// Details an individual cluster's current status.
+message ClusterStatus {
+  // Name of the cluster.
+  string name = 1;
+
+  // General outlier statistics if installed for this cluster.
+  OutlierInfo outlier_info = 2;
+
+  // Denotes whether this cluster was added via API or configured statically.
+  bool added_via_api = 3;
+
+  // Mapping from host address to the host's current status.
+  repeated HostStatus host_statuses = 4;
+}
+
+// :ref:`Cluster outlier detection <arch_overview_outlier_detection>` statistics. -1 for any of
+// these statistics denotes that there was not enough data to compute it.
+message OutlierInfo {
+  // The average success rate of the hosts in the Detector for the last aggregation interval.
+  //
+  // Note: the message will be omitted if there were not enough hosts with enough request volume to
+  // proceed with success rate based outlier ejection.
+  envoy.type.Percent success_rate_average = 1;
+
+  // The success rate threshold used in the last interval. The threshold is used to eject hosts
+  // based on their success rate.
+  //
+  // Note: the message will be omitted if there were not enough hosts with enough request volume to
+  // proceed with success rate based outlier ejection. Since this value can technically be negative
+  // due to the way that it's computed, negative values will also be represented by the absence of
+  // this field because a negative value implies that there was no threshold for that interval.
+  envoy.type.Percent success_rate_ejection_threshold = 2;
+}
+
+// Current state of a particular host.
+message HostStatus {
+  // Address of this host.
+  envoy.api.v2.core.Address address = 1;
+
+  // Mapping from the name of the statistic to the current value.
+  map<string, int64> stats = 2;
+
+  // The host's current health status.
+  HostHealthStatus health_status = 3;
+
+  // Configured load balancing weight for this host.
+  uint64 weight = 4;
+
+  // Configured locality for the host.
+  envoy.api.v2.core.Locality locality = 5;
```

Comment:
Can we reuse https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/endpoint/endpoint.proto#L46 and related structures? Seems like a reasonable overlap.

Reply:
Hmmmm, yes, there's definitely some overlap, but I worry that this will creep into us providing the entire EDS config in `/clusters` (the overlap isn't 100%, so we would have to leave out fields).  Didn't you have a suggestion on the previous round of review that we should eliminate config details that could easily be found in `/config_dump` to delineate status from configuration?  Given that EDS info should be available in `/config_dump`, should we try to remove all of that from the host portion as well?  I'm not super opinionated, but I think our approach should be consistent.

Reply:
`/config_dump` doesn't provide EDS today :) @mattklein123 do you think we should provide EDS here or in `/config_dump`? I think the latter, but this PR seems related.

Reply:
Yeah I think it should be provided in `/config_dump`. The reason I didn't do it is that it isn't a trivial change/design to figure out how to do it. Should it be top level? Should it be embedded within clusters? How will the config source be managed? Etc. I opened the issue to track adding it and hopefully we can discuss there when someone wants to work on it.

Reply:
SGTM, I'll remove `weight`, `locality`, and `canary` from `HostStatus`.